specifyweb/frontend/js_src/lib/components/QueryBuilder/ResultsTable.tsx

```diff
@@ -178,7 +177,7 @@ function Row({
         </div>
       )}
       {result
-        .filter((_, index) => !hasIdField || index !== queryIdField)
+        .filter((_, index) => index !== queryIdField)
```

Comment:
I think there are more places in the query builder that change their logic based on whether hasIdField or not
@CarolineDenis @acwhite211  would either of you be willing to check if there are more possible places so that we can also refactor them? in general, there shouldn't be any references to `hasIdField` left, right?